### ISSUE TITLE
scikit-build does not work with pip 22.2 in edit mode

### ISSUE THREAD

**Issue by njzjz** [scikit-build/scikit-build#740]
In pip 22.1, when installing with `pip install -v -e .`, CMake will be executed.

In pip 22.2, using the same command, CMake won't be executed during installation.

**Comment by njzjz** [comment]
A temporary solution is to set the environment variable:
```sh
export SETUPTOOLS_ENABLE_FEATURES="legacy-editable"
```

It looks like setuptools 64 changes the behavior of the editable mode.

**Comment by henryiii** [comment]
Yes, this was seen by @kratsg shortly after 64 was released. That's a fine workaround until we figure out what needs to be done to support the new editable mode. I'll be more familiar with what needs to be done after working on scikit_build_core for a while, so probably will focus there unless someone comes along sooner to do it.

**Comment by casperdcl** [comment]
running into the same problem.

Arrived here via https://github.com/pypa/setuptools/issues/3550 <- https://github.com/pypa/setuptools/pull/3488 <- https://github.com/AMYPAD/CuVec/pull/25

**Comment by abravalheri** [comment]
Just in case anyone is looking into this problem, the documentation for custom build was updated in the v64 release: https://setuptools.pypa.io/en/latest/userguide/extension.html. Hopefully this is useful.

**Comment by 0x2b3bfa0** [comment]
#### Related:
* https://github.com/pypa/setuptools/pull/3414

**Comment by henryiii** [comment]
I believe this requires implementing `get_output` and `get_output_mapping`. This is something the current scikit-build can't produce, since it delegates the build to CMake, and has no idea about what is going in and out. `scikit-build-core`, however, does know this (via requiring CMake 3.15+ and using the File API introduced in 3.15), so I think this will have to wait until scikit-build-core is ready.

Unless I'm mistaken, but I think that's what is required.

**Comment by henryiii** [comment]
scikit-build (classic) after #1185 is now a thin wrapper around scikit-build-core's setuptools plugin. Closing all open issues; if this still applies, please reopen or open an issue on scikit-build-core. Thanks!
